src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -1644,14 +1644,31 @@ private static TypeSymbol GetParameterType(int argIndex, MemberAnalysisResult re
                 }
             }
 
-            // Finally, if one has fewer custom modifiers, that is better
+            // Otherwise, if one has fewer custom modifiers, that is better
             int m1ModifierCount = m1.LeastOverriddenMember.CustomModifierCount();
             int m2ModifierCount = m2.LeastOverriddenMember.CustomModifierCount();
             if (m1ModifierCount != m2ModifierCount)
             {
                 return (m1ModifierCount < m2ModifierCount) ? BetterResult.Left : BetterResult.Right;
             }
 
+            // Otherwise, prefer methods with 'val' parameters over 'in' parameters.
+            Debug.Assert(m1Original.Length == m2Original.Length, "Unequal parameters counts are handled in an earlier stage");
+            for (i = 0; i < m1Original.Length; i++)
+            {
```

Comment:
It seems the order of paremeters matters. 
I.E. we can resolve M(1, 1) when
M(in int x, int y)
M(int x, in int y)

Second wins. Is that intentional?